### ISSUE TITLE
Found with fuzzer:  invalid material 14 (otyp 107) obj 60c000000400 invent: 38 +0 silver bullets (in quiver pouch)

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#13]
```
Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
#2  0x0000555555a44c60 in NH_abort () at end.c:236
#3  0x0000555555a4ba5c in panic (str=0x55555651b6c0 "%s") at end.c:802
#4  0x0000555555f98c03 in impossible (s=0x5555564d0b00 <ofmt0> "%s obj %s %s: %s")
    at pline.c:518
#5  0x0000555555d53cd5 in insane_object (obj=0x60c000000400, 
    fmt=0x5555564d0b00 <ofmt0> "%s obj %s %s: %s", 
    mesg=0x7fffffffe110 "invalid material 14 (otyp 107)", mon=0x0) at mkobj.c:2787
#6  0x0000555555d53419 in objlist_sanity (objlist=0x60c000000280, wheretype=3, 
    mesg=0x5555564d0d00 "invent sanity") at mkobj.c:2705
#7  0x0000555555d52ac0 in obj_sanity_check () at mkobj.c:2624
#8  0x000055555585faa3 in sanity_check () at cmd.c:4840
#9  0x0000555555761767 in moveloop (resuming=0 '\000') at allmain.c:767
#10 0x00005555563b4127 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737337632192, 93824997411122, 335544320, 
            140737341055120, 1073741824, 0, 140737342801568, 4294967295, 140737337360384, 
            140737342784672, 9438383760701890816}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342784672, 9438383760701890816, 
              140737488343888, 140737342801568, 18446744073709547520, 140737327112192, 
              140737345470680, 140737488344960, 140737488344960, 140737344339986, 
              17592186042986, 140737488338944, 140737341055120, 9438383760701890816, 
              140737488344288}}, sa_flags = 242458880, sa_restorer = 0x7fffffffd6a0}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a44c60 in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a4ba5c in panic (str=0x55555651b6c0 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffd770, 
            reg_save_area = 0x7fffffffd6a0}}
#4  0x0000555555f98c03 in impossible (s=0x5555564d0b00 <ofmt0> "%s obj %s %s: %s")
    at pline.c:518
        pbuf = "invalid material 14 (otyp 107) obj 60c000000400 invent: 38 +0 silver bullets (in quiver pouch)\000\000\000\000\000\000\000\000\000\000\fY_\367\377\177", '\000' <repeats 90 times>...
        the_args = {{gp_offset = 40, fp_offset = 48, overflow_arg_area = 0x7fffffffde60, 
            reg_save_area = 0x7fffffffdd80}}
#5  0x0000555555d53cd5 in insane_object (obj=0x60c000000400, 
    fmt=0x5555564d0b00 <ofmt0> "%s obj %s %s: %s", 
    mesg=0x7fffffffe110 "invalid material 14 (otyp 107)", mon=0x0) at mkobj.c:2787
        objnm = 0x5555567c618a <obufs+842> "38 +0 silver bullets (in quiver pouch)"
        monnm = 0x5555564d1380 "null!"
        altfmt = "\020\337\377\377\377\177\000\000@\036\000\000\300`", '\000' <repeats 18 times>, "\263\212\265A\000\000\000\000\000\024MVUU\000\000+C\325UUU", '\000' <repeats 26 times>, "containe\000\004\000\000\300`\000\000sanity\000\000\000\241s\016\006\340\373\202\000\000\000\000\000\000\000\000\022\374\377\377\377\017\000\000\220\340\377\377\377\177\000\000p\342\377\377\377\177\000\000\220\340\377\377\377\177\000\000\000\000\000\000\000\000\000\000\220\342\377\377\377\177\000\000\337\313b\367\377\177", '\000' <repeats 18 times>, " \000\000\000\060\000\000\000"...
#6  0x0000555555d53419 in objlist_sanity (objlist=0x60c000000280, wheretype=3, 
    mesg=0x5555564d0d00 "invent sanity") at mkobj.c:2705
        matbuf = "invalid material 14 (otyp 107)", '\000' <repeats 225 times>
        maskbuf = '\000' <repeats 39 times>
        bc_ok = 0 '\000'
        obj = 0x60c000000400
#7  0x0000555555d52ac0 in obj_sanity_check () at mkobj.c:2624
        x = 80
        y = 21
        obj = 0x0
#8  0x000055555585faa3 in sanity_check () at cmd.c:4840
No locals.
#9  0x0000555555761767 in moveloop (resuming=0 '\000') at allmain.c:767
        moveamt = 10
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 30867
        past_clock = -30865
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#10 0x00005555563b4127 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
For now I have disabled silver bullets and silver daggers for undead slayers.

**Comment by elunna** [comment]
Fixed in commit cf5afef15: Added fix to setup undead slayers with silver daggers/bullets regardless of race.
